pkg/scheduler/plugins/elasticquota/plugin.go

```diff
@@ -0,0 +1,560 @@
+			g.groupQuotaManager.UpdateGroupDeltaUsed(newQuotaName, podReq)
+			g.quotaOverUsedRevokeController.addPodWithQuotaName(newQuotaName, newPod)
+		}
+	}
+
+	klog.V(5).Infof("OnPodUpdateFunc %v.%v add request success, quotaName:%v", newPod.Namespace, newPod.Name, newQuotaName)
+}
+
+func (g *Plugin) OnPodDelete(obj interface{}) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return
+	}
+
+	g.Lock()
+	defer g.Unlock()
+
+	quotaName := g.getPodAssociateQuotaName(pod)
+	if err := g.elasticQuotaInfos[quotaName].deletePodIfPresent(pod); err != nil {
+		// the pod is not added to the g.elasticQuotaInfos[quotaName], so no need to call UpdateGroupDeltaUsed.
+		return
+	}
+	podReq, _ := resource.PodRequestsAndLimits(pod)
+	podReqTmp := quotav1.Subtract(corev1.ResourceList{}, podReq)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(quotaName, podReqTmp)
+
+	g.groupQuotaManager.UpdateGroupDeltaUsed(quotaName, podReqTmp)
+	g.quotaOverUsedRevokeController.erasePodWithQuotaName(quotaName, pod)
+	klog.V(5).Infof("OnPodDeleteFunc %v.%v delete request success", pod.Namespace, pod.Name)
+}
+
+func (g *Plugin) OnNodeAdd(obj interface{}) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		return
+	}
+
+	klog.V(5).Infof("OnNodeAddFunc add node[%v]", node.Name)
+	if node.DeletionTimestamp != nil {
+		klog.Errorf("node is deleting:%v", node.Name)
+		return
+	}
+
+	allocatable := node.Status.Allocatable
+
+	g.nodeResourceMapLock.Lock()
+	defer g.nodeResourceMapLock.Unlock()
+	if _, ok := g.nodeResourceMap[node.Name]; ok {
+		klog.Infof("OnNodeAddFunc skip due OnAdd Twice:%v", node.Name)
+		return
+	} else {
+		g.nodeResourceMap[node.Name] = struct{}{}
+	}
+
+	g.groupQuotaManager.UpdateClusterTotalResource(allocatable)
+}
+
+func (g *Plugin) OnNodeUpdate(oldObj, newObj interface{}) {
+	newNode := newObj.(*corev1.Node)
+	oldNode := oldObj.(*corev1.Node)
+
+	g.nodeResourceMapLock.Lock()
+	defer g.nodeResourceMapLock.Unlock()
+
+	if _, exist := g.nodeResourceMap[newNode.Name]; !exist {
+		return
+	}
+
+	if newNode.ResourceVersion == oldNode.ResourceVersion {
+		klog.Warningf("update node warning, update version for the same, nodeName:%v", newNode.Name)
+		return
+	}
+
+	klog.V(5).Infof("OnNodeUpdateFunc update:%v delete:%v", newNode.Name, newNode.DeletionTimestamp)
+
+	if newNode.DeletionTimestamp != nil {
+		return
+	}
+
+	oldNodeAllocatable := oldNode.Status.Allocatable
+	newNodeAllocatable := newNode.Status.Allocatable
+
+	if quotav1.Equals(oldNodeAllocatable, newNodeAllocatable) {
+		return
+	}
+
+	deltaNodeAllocatable := quotav1.Subtract(newNodeAllocatable, oldNodeAllocatable)
+	g.groupQuotaManager.UpdateClusterTotalResource(deltaNodeAllocatable)
+	klog.V(5).Infof("OnNodeUpdateFunc success:%v [%v]", newNode.Name, newNodeAllocatable)
+}
+
+func (g *Plugin) OnNodeDelete(obj interface{}) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		klog.Errorf("node is nil")
+		return
+	}
+
+	allocatable := node.Status.Allocatable
+	klog.V(5).Infof("OnNodeDeleteFunc:%v", node.Name)
```

Comment:
klog.V(5).Infof("OnNodeDeleteFunc:%v", node.Name) should be killed

Reply:
fixed